app/services/preserved_object_handler.rb

```diff
@@ -133,18 +133,25 @@ def update_version
 
   private
 
-  def create_db_objects(status)
+  def create_db_objects(status, validate=false)
     results = []
     pp_default = PreservationPolicy.default_preservation_policy
     create_results = with_active_record_rescue do
       po = PreservedObject.create!(druid: druid,
                                    current_version: incoming_version,
                                    preservation_policy: pp_default)
-      PreservedCopy.create!(preserved_object: po,
-                            version: incoming_version,
-                            size: incoming_size,
-                            endpoint: endpoint,
-                            status: status)
+      pc = PreservedCopy.create!(preserved_object: po,
+                                 version: incoming_version,
+                                 size: incoming_size,
+                                 endpoint: endpoint,
+                                 status: status)
+      if validate
+        t = Time.current
+        # Returns the value of time as an integer number of seconds since the Epoch.
+        pc.last_audited = t.to_i
+        pc.last_checked_on_storage = t
+        pc.save
+      end
```

Comment:
what about something like:
```ruby
pc_attrs = {
  preserved_object: po,
  version: incoming_version,
  size: incoming_size,
  endpoint: endpoint,
  status: status
}

if validate
  t = Time.current
  # Returns the value of time as an integer number of seconds since the Epoch.
  pc_attrs.merge({
    last_audited: t.to_i,
    last_checked_on_storage: t
  })
end

PreservedCopy.create!(pc_attrs)
```

and then you save yourself a possible extra trip to the database, since there's no optional update, just a create all at once, with some optional attributes.